javalib/src/main/scala/java/util/concurrent/ThreadLocalRandom.scala

```diff
@@ -0,0 +1,98 @@
+/*
+ * closely translated from:
+ * https://github.com/scala/scala
+ * scala.concurrent.fokjoin.ThreadLocalRandom
+ */
+
+package java.util.concurrent
+
+import java.util.Random
+
+class ThreadLocalRandom extends Random {
+
+  override protected def _setSeed(seed_in: Long): Unit = ()
+
+  override def setSeed(seed: Long): Unit =
+    throw new UnsupportedOperationException()
+
+  /*
+  * this code has to be removed!!!
+  */
+  import scala.scalajs.js
+  /** Generate a random long from JS RNG to seed a new Random */
+  private def randomSeed(): Long = {
+    val h = randomInt().toLong
+    val l = randomInt().toLong
+    (h << 32) | (l & 0xffffffffL)
+  }
+
+  private def randomInt(): Int =
+    (Math.floor(js.Math.random() * 4294967296.0) - 2147483648.0).toInt
+  /*
+  * end of code to be removed
+  */
+
+
+  def nextInt(least: Int, bound: Int): Int = {
+    if (least >= bound) throw new IllegalArgumentException()
+
+    //nextInt(bound - least) + least
+    (nextLong(bound.toLong - least.toLong) + least.toLong).toInt
```

Comment:
Ah. Then I would suggest to optimize a fast path for cases where no overflow occurs, because Longs are really really slow:
```scala
val diff = bound - least
if (diff > 0) nextInt(diff) + least
else (nextLong(bound.toLong - least.toLong) + least.toLong).toInt
```